sortinghat/core/db.py

```diff
@@ -1206,3 +1207,135 @@ def move_team(trxl, team, organization):
                        target=op_args['team'])
 
     return team
+
+
+def add_scheduled_task(trxl, job_type, interval=None, args=None, job_id=None):
+    """Add an scheduled task to the database.
+
+    This function adds a new task to the database.
+
+    As a result, the function returns a new `ScheduledTask` object.
+
+    :param trxl: TransactionsLog object from the method calling this one
+    :param job_type: name of the job to be scheduled
+    :param interval: period of executions, in minutes. None to disable
+    :param args: specific arguments for the job
+    :param job_id: current job running the task
+
+    :returns: a new ImportIdentitiesTask
+    """
+    # Setting operation arguments before they are modified
+    op_args = {
+        'job_type': job_type,
+        'interval': str(interval),
+        'args': str(args),
+        'job_id': job_id,
+    }
+
+    validate_field('job_type', job_type)
+
+    if interval and interval < 0:
+        raise ValueError("'interval' must be a positive number or 0.")
+
+    try:
+        task = ScheduledTask(
+            job_type=job_type,
+            interval=interval,
+            args=args,
+            job_id=job_id
+        )
+        task.save()
+    except django.db.utils.IntegrityError as exc:
+        _handle_integrity_error(ScheduledTask, exc)
+
+    trxl.log_operation(op_type=Operation.OpType.ADD, entity_type='scheduled_task',
+                       timestamp=datetime_utcnow(), args=op_args,
+                       target=op_args['job_type'])
+
+    return task
+
+
+def update_scheduled_task(trxl, task, **kwargs):
+    """Update an import identities task.
+    This function allows to edit or update the information of
+    the given task. The values to update are given as keyword
+    arguments. The allowed keys are listed below
+    (other keywords will be ignored):
+       - `backend`: name of the importer backend
+       - `url`: URL of a file or API to fetch the identities from
+       - `interval`: period of executions, in minutes. None to disable
+       - `params`: specific parameters for the importer backend
+    As a result, it will return the `ImportIdentitiesTask` object with
+    the updated data.
+    :param trxl: TransactionsLog object from the method calling this one
+    :param task: task to update
+    :param kwargs: parameters to edit the task
+    :returns: task object with the updated information
+    :raises ValueError: raised when an invalid value is provided
+    """
```

Comment:
```suggestion
    """Update an import identities task.
    
    This function allows to edit or update the information of
    the given task. The values to update are given as keyword
    arguments. The allowed keys are listed below
    (other keywords will be ignored):
       - `backend`: name of the importer backend
       - `url`: URL of a file or API to fetch the identities from
       - `interval`: period of executions, in minutes. None to disable
       - `params`: specific parameters for the importer backend
    
    As a result, it will return the `ImportIdentitiesTask` object with
    the updated data.
    
    :param trxl: TransactionsLog object from the method calling this one
    :param task: task to update
    :param kwargs: parameters to edit the task
    
    :returns: task object with the updated information
    
    :raises ValueError: raised when an invalid value is provided
    """
```